### PR TITLE
When Recurrence meets Transformers to keras 3.0

### PR DESCRIPTION
**Pull request by chunduriv**
This PR changes the When Recurrence meets Transformers to keras 3.0, as requested in https://github.com/keras-team/keras-cv/issues/2211

Please review the attached [gist](https://colab.sandbox.google.com/gist/chunduriv/5c64a264737b25df4d7050079d5b6862/temporal_latent_bottleneck_v3.ipynb)

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! I'm not seeing any usage of TF in the example, so does it run on JAX and torch?

> The patching operation is done using a `keras.layers.Conv2D` instance instead of a
traditional `tf.image.extract_patches` to allow for vectorization.

Please update this -- no reference to TF needed.

**Comment by chunduriv** [comment]
@fchollet, thanks for the review. 

> so does it run on JAX and torch?

It now supports all backends.

> Please update this -- no reference to TF needed.

Done.

File: examples/vision/temporal_latent_bottleneck.py [hunk 1]
```diff
@@ -43,29 +43,25 @@
...
 ## Setup imports
 """
+import os
 
-import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import layers
-from tensorflow.keras import mixed_precision
-from tensorflow.keras.optimizers import AdamW
+os.environ["KERAS_BACKEND"] = "tensorflow"
```
Comment (fchollet) [inline examples/vision/temporal_latent_bottleneck.py:53]: If it supports all backends, please remove the line that sets the backend.

**Review by fchollet — APPROVED** [review]
Looks great -- please add the generated ipynb and md files!

File: examples/vision/md/temporal_latent_bottleneck.md [hunk 2]
```diff
@@ -925,6 +981,12 @@ index = random.randint(0, config["batch_size"])
 orig_image = images[index]
 overlay_image = upsampled_heat_map[index, ..., 0]
 
+if keras.backend.backend() == "torch":
```
Comment (fchollet) [inline examples/vision/md/temporal_latent_bottleneck.md:984]: Please note, if you use `ops.convert_to_numpy()` you don't need any torch specific code. This is not a blocker so we can merge regardless.

**Pull request merged**